lib/internal/process.js

```diff
@@ -65,6 +65,22 @@ function setupConfig(_source) {
     if (value === 'false') return false;
     return value;
   });
+  // Intl.v8BreakIterator() without data crashes with a fatal error, so throw instead.
+  if (global.hasOwnProperty('Intl') &&
+      Intl.hasOwnProperty('v8BreakIterator') &&
+      process.hasOwnProperty('config') &&
+      process.config.hasOwnProperty('variables') &&
+      process.config.variables.icu_small &&
+      !process.icu_data_dir) {
```

Comment:
Well, there's currently no other option. I'm going to be working up a PR that provides an internal version of this we can rely on in case userland does overwrite `process.config`. This can be updated once that lands.